### PR TITLE
feat: add useCardAnimation

### PR DESCRIPTION
**Pull request by adamgrzybowski**
**Motivation**

Make animation progress available inside a screen.

**Test plan**

Describe the **steps to test this change** so that a reviewer can verify it. Provide screenshots or videos if the change affects UI.

The change must pass lint, typescript and tests.

### CODE REVIEW

File: packages/material-top-tabs/src/utils/useTabAnimation.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+import { TabAnimationContext } from './TabAnimationContext';
+
+export function useTabAnimation() {
+  const animation = React.useContext(TabAnimationContext);
+
+  if (animation === undefined) {
+    throw new Error(
+      "Couldn't find values for card animation. Are you inside a screen in Tab?"
```
Comment (osdnk) [inline packages/material-top-tabs/src/utils/useTabAnimation.tsx:10]: i believe it's worth clarifying that it's material-top-tabs (and not bottom-tabs).

**Review by osdnk — CHANGES_REQUESTED** [review]
Thanks for those changes. I think I need to suggest a slightly different API. In my opinion, we should target a similar API as `useCardAnimation`.

Particularly, I believe that in the future we may support multiple animations. In the same way, as we recently merged #11323, we can also imagine a similar animation inside `material-top-tabs`. 

Therefore, I would suggest making the API more open for multiple animations and returning an object, instead of just a single animated value. The current value may live under the `position` field in the object. With this approach, we will not be forced to make a breaking change (or design a weird API).

`const { position } = useTabAnimation()`

File: packages/material-top-tabs/src/utils/useTabAnimation.tsx (hunk 1 above)
Comment (osdnk) [reply packages/material-top-tabs/src/utils/useTabAnimation.tsx:10]: Also, `card animation`? 😅

**Comment by adamgrzybowski** [comment]
Hey @osdnk, I made some changes. Please take a look when you have a moment

**Review by osdnk — CHANGES_REQUESTED** [review]
I think maybe i was not clear about that. I think that the name `useCardAnimation` is not OK here - we don't have any card. `useTabAnimation` is ok.

`Couldn't find values for tab animation. Are you inside a screen in Material Top Tab navigator?` should be the message

**Comment by adamgrzybowski** [comment]
@osdnk Yeah I can see it now 😄 idk why I understood it wrong. I pushed fixed version.

**Review by osdnk — APPROVED** [review]

**Pull request merged**